src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/ColorTransform.cs

```diff
@@ -170,21 +148,20 @@ private void InitializeICM()
             _colorTransformHelper = new ColorTransformHelper();
         }
 
-        private long ICM2Color(float[] srcValue)
+        private long ICM2Color(Span<float> srcValue)
         {
             long colorValue;
 
-            if (srcValue.GetLength(0) < 3 || srcValue.GetLength(0) > 8)
+            if (srcValue.Length < 3 || srcValue.Length > 8)
             {
                 throw new NotSupportedException(); // Only support color spaces with 3,4,5,6,7,8 channels
             }
 
-            if (srcValue.GetLength(0) <= 4)
+            if (srcValue.Length <= 4)
             {
-                UInt16[] channel = new UInt16[4];
+                Span<UInt16> channel = stackalloc UInt16[4];
 
```

Comment:
Same I would suggest here - 
channel.Clear();

Reply:
The `src.Value` is gauranteed to be 4 here. All `channel` values are assigned below.

Reply:
But here the check says that it could be less that 4 too.
if (srcValue.Length <= 4)

Reply:
Oh sorry it can also be 3, my bad, you are correct!